src/scheduler/pbs_bitmap.c

```diff
@@ -0,0 +1,346 @@
+	long local_bit_ind;
+	int i;
+	
+	if (pbm == NULL)
+		return -1;
+	
+	if (start_bit >= pbm->num_bits)
+		return -1;
+	
+	long_ind = start_bit / BYTES_TO_BITS(sizeof(unsigned long));
+	local_bit_ind = start_bit % BYTES_TO_BITS(sizeof(unsigned long));
+
+	/* special case - look at first long that contains start_bit */
+	if (pbm->bits[long_ind] != 0) {
+		for (i = local_bit_ind + 1; i < BYTES_TO_BITS(sizeof(unsigned long)); i++) {
+			if (pbm->bits[long_ind] & (1UL << i)) {
+				return (long_ind * BYTES_TO_BITS(sizeof(unsigned long)) + i);
+			}
+		}
+
+		/* didn't find an on bit after start_bit_index in the long that contained start_bit */
+		if (long_ind < pbm->num_longs) {
+			long_ind++;
+		}
+	}
+
+	for( ; long_ind < pbm->num_longs && pbm->bits[long_ind] == 0; long_ind++)
+		;
+
+	if (long_ind == pbm->num_longs)
+		return -1;
+
+	for (i = 0 ; i < BYTES_TO_BITS(sizeof(unsigned long)) ; i++) {
+		if (pbm->bits[long_ind] & (1UL << i)) {
+			return (long_ind * BYTES_TO_BITS(sizeof(unsigned long)) + i);
+		}
+	}
+
+	return -1;
+}
+
+/**
+ * @brief get the first on bit
+ * @param bm - the bitmap
+ * @return int
+ * @retval the bit number of the first on bit
+ * @retval -1 on error
+ */
+int
+pbs_bitmap_first_bit(pbs_bitmap *bm)
+{
+	if(pbs_bitmap_get_bit(bm, 0))
+		return 0;
+	
+	return pbs_bitmap_get_next_on_bit(bm, 0);
+}
+
+/**
+ * @brief pbs_bitmap version of L = R
+ * @param L - bitmap lvalue
+ * @param R - bitmap rvalue
+ * @return int
+ * @retval 1 success
+ * @retval 0 failure
+ */
+int
+pbs_bitmap_equals(pbs_bitmap *L, pbs_bitmap *R)
+{
+	int i;
+	
+	if (L == NULL || R == NULL)
+		return 0;
+	
+	/* In the case where R is longer than L, we need to allocate more space for L
+	 * Instead of using R->num_bits, we call pbs_bitmap_alloc() with the 
+	 * full number of bits required for its num_longs.  This is because it
+	 * is possible that R has more space allocated to it than required for its num_bits.
+	 * This happens if it had a previous call to pbs_bitmap_equals() with a shorter bitmap.
+	 */
+	if (R->num_longs > L->num_longs)
+		if(pbs_bitmap_alloc(L, BYTES_TO_BITS(R->num_longs * sizeof(unsigned long))) == NULL)
+			return 0;
+	
+	for(i = 0; i < R->num_longs; i++)
+		L->bits[i] = R->bits[i];
+	if (R->num_longs < L->num_longs)
+		for(; i < L->num_longs; i++)
+			L->bits[i] = 0;
+	
+	L->num_bits = R->num_bits;
+	return 1;
+}
+
+/**
+ * @brief pbs_bitmap version of L == R
+ * @param L - bitmap lvalue
+ * @param R - bitmap rvalue
+ * @return int
+ * @retval 1 bitmaps are equal
+ * @retval 0 bitmaps are not equal
+ */
+int
+pbs_bitmap_is_equal(pbs_bitmap *L, pbs_bitmap *R)
+{
+	int i;
+	
+	if(L == NULL || R == NULL)
+		return 0;
+	
+	if (L->num_bits != R->num_bits)
+		return 0;
+	
+	for(i = 0; i < L->num_longs; i++)
+		if(L->bits[i] != R->bits[i])
+			return 0;
+	
+	return 1;
+}
+
+/**
+ * @brief count the number of on bits
+ * @param bm - the bitmap
+ * @return int
+ * @retval number of on bits
+ */
+int
+pbs_bitmap_count(pbs_bitmap *bm)
```

Comment:
Since this is only counting the bits which are on, please rename this to pbs_bitmap_count_on.